src/format/mod.rs

```diff
@@ -17,20 +17,54 @@
 
 #![allow(ellipsis_inclusive_range_patterns)]
 
-use std::fmt;
-use std::str::FromStr;
+use core::fmt;
+use core::str::FromStr;
+#[cfg(any(feature = "std", test))]
 use std::error::Error;
+#[cfg(any(feature = "alloc", feature = "std", test))]
+use alloc::boxed::Box;
+#[cfg(any(feature = "alloc", feature = "std", test))]
+use alloc::string::{String, ToString};
+
+#[cfg(not(any(feature = "alloc", feature = "std", test)))]
+mod core_only {
+    /// Core only
+    #[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+    pub struct Box<T: ?Sized>(core::marker::PhantomData<T>);
```

Comment:
This thing is kind of hacky, but it allows at least partially supporting some of the format module stuff.

Reply:
Can you explain how/if this actually works/what the runtime experience would be? Or is there some explanation of this pattern somewhere?

Alternatively, would you mind adding some small code to the core test file that demonstrates what this ends up doing?

Reply:
It's only so the Item enum actually compiles as there's Box<str> in there. I'm not sure if we even need this enum when alloc isn't compiled in. I honestly don't know what even works in that module at all if DelayedFormat isn't compiled in.

Reply:
Does the timezone for the DelayedFormat really need to be a String or could it be turned into a &'static str? (I'm not sure if there ever are any truly dynamic timezones that would need String)

Reply:
Wait, what is OwnedLiteral and OwnedSpace even used for? I don't see them being used anywhere.

Reply:
Yes I ended up putting these variants behind the same feature gate, I'm not sure that it will have any effect on usability. _Potentially_ if people were compiling their own `format::Item` lists it could have been a thing?